draft-kazuho-quic-authenticated-handshake.md

```diff
@@ -340,26 +340,31 @@ packet means that the retransmitted Initial packets would become undecryptable
 and therefore be deemed lost by the client, reducing the client's congestion
 window size.
 
-## No Support for Split Mode
-
-Under the design discussed in this document, it is impossible to use an
-unmodified QUIC server as a backend server in "Split Mode" ([TLS-ESNI];
-section 3) due to the following two reasons:
-
-* Access to initial_auth_secret is required for generating and validating
-  Initial packets.  However, the backend server, not knowing the ESNI private
-  key, cannot calculate the secret.
-
-* The client-facing server cannot continue forwarding packets to the correct
-  destination when there is a change in Connection ID mid-connection.
-
-To address the issues, we might consider specifying a protocol that will be
-used between the client-facing server and the backend server for communicating
-the initial_auth_secret and the spare Connection IDs.  Note that such protocol
-can be lightweight, assuming the communication between the two servers will be
-over a virtual private network.  Such assumption can be made because the
-backend server cannot operate QUIC without access to the source address-port
-tuple of the packets that the client has sent.
+## Split Mode
+
+To support server-side deployments using "Split Mode" ([TLS-ESNI]; section 3),
+the following properties need to be exchanged between the fronting server and
+the hidden server, in addition to those required by the Encrypted SNI
+extension:
+
+* hmac_key
+* spare Connection IDs
+* ODCID
+
+Both the fronting server and the hidden server need access to the hmac_key to
+authenticate the Initial packets.  However, because the key is derived from
+the shared DH secret of ESNI, it is not necessarily available to the hidden
+server.
+
+Access to spare Connection IDs is mandatory to support clients migrating to
+different addresses.  The fronting server, without access to the 1-RTT QUIC
+frames being exchanged, cannot track the mapping of the Connection IDs that
+change mid-connection.
+
+ODCID is necessary to decrypt an Initial packet sent in response to a Retry.
+However, the value is typically available only to the server that generates
+the Retry.  The fronting server and the hidden server need to exchange the
+ODCID, or provide the secret for extracting the ODCID from a Retry token.
 
```

Comment:
Split mode retry also requires passing the ODCID. That should be a fairly common feature.

Reply:
I agree, and I think that is covered by the text above.

FWIW, my understanding is that either the fronting server or the hidden server might generate a Retry, and I've tried to cover the two cases in the text being proposed.